### DISCUSSION TITLE
How to configure new line indentation?

### DISCUSSION THREAD

**Discussion by wojdyr** [zed-industries/zed#21135] [Feature Requests]
The indentation I get from Ctrl-Shift-I formatting and the indentation on a new line or when pasting text is inconsistent for C++. Is this because one is handled by clangd and the other by tree-sitter?
When formatting, there is no indentation after `namespace {` (I could change it in .clang-format):
```cpp
namespace {
something...
```
but pressing Enter or pasting code after `namespace {` adds one (unwanted) indentation level.
Is there a way to disable it?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
